pkg/gardenlet/controller/seed/seed_care_control.go

```diff
@@ -0,0 +1,209 @@
+		conditions := buildSeedConditions(seed.Status.Conditions, updatedConditions, conditionTypes)
+		log.Info("[SEED CARE] Updating seed status conditions and constraints")
+		if err := patchSeedStatus(ctx, gardenClient, seed, conditions); err != nil {
+			log.Error(err, "Could not update Seed status")
+			return nil // We do not want to run in the exponential backoff for the condition checks.
+		}
+	}
+	return nil
+}
+
+func careSetupFailure(ctx context.Context, gardenClient client.Client, seed *gardencorev1beta1.Seed, message string, conditions []gardencorev1beta1.Condition) error {
+	updatedConditions := make([]gardencorev1beta1.Condition, 0, len(conditions))
+	for _, cond := range conditions {
+		updatedConditions = append(updatedConditions, gardencorev1beta1helper.UpdatedConditionUnknownErrorMessage(cond, message))
+	}
+
+	if !gardencorev1beta1helper.ConditionsNeedUpdate(conditions, updatedConditions) {
+		return nil
+	}
+
+	return patchSeedStatus(ctx, gardenClient, seed, updatedConditions)
+}
+
+// buildSeedConditions builds and returns the seed conditions using the given seed conditions as a base,
+// by first removing all conditions with the given types and then merging the given conditions (which must be of the same types).
+func buildSeedConditions(seedConditions []gardencorev1beta1.Condition, conditions []gardencorev1beta1.Condition, conditionTypes []gardencorev1beta1.ConditionType) []gardencorev1beta1.Condition {
+	result := gardencorev1beta1helper.RemoveConditions(seedConditions, conditionTypes...)
+	result = gardencorev1beta1helper.MergeConditions(result, conditions...)
+	return result
+}
+
+func patchSeedStatus(ctx context.Context, c client.StatusClient, seed *gardencorev1beta1.Seed, conditions []gardencorev1beta1.Condition) error {
+	patch := client.StrategicMergeFrom(seed.DeepCopy())
+	seed.Status.Conditions = conditions
+	return c.Status().Patch(ctx, seed, patch)
+}
+
+func (r *careReconciler) conditionThresholdsToProgressingMapping() map[gardencorev1beta1.ConditionType]time.Duration {
+	out := make(map[gardencorev1beta1.ConditionType]time.Duration)
+	for _, threshold := range r.config.Controllers.SeedCare.ConditionThresholds {
+		out[gardencorev1beta1.ConditionType(threshold.Type)] = threshold.Duration.Duration
+	}
+	return out
+}
+
+func (c *Controller) seedCareAdd(obj interface{}) {
+	key, err := cache.MetaNamespaceKeyFunc(obj)
+	if err != nil {
+		return
+	}
+	c.seedCareQueue.Add(key)
+}
+
+func (c *Controller) seedCareUpdate(oldObj, newObj interface{}) {
+	key, err := cache.MetaNamespaceKeyFunc(newObj)
+	if err != nil {
+		return
+	}
+
+	oldSeed, ok := oldObj.(*gardencorev1beta1.Seed)
+	if !ok {
+		return
+	}
+	newSeed, ok := newObj.(*gardencorev1beta1.Seed)
+	if !ok {
+		return
+	}
+
+	specChanged := !apiequality.Semantic.DeepEqual(oldSeed.Spec, newSeed.Spec)
+	statusChanged := !apiequality.Semantic.DeepEqual(oldSeed.Status, newSeed.Status)
+
+	if !specChanged && statusChanged {
```

Comment:
If `specChanged`==false and `statusChanged==false` the seed will be added to the queue, this means that `metadata` changes will trigger care checks. 
Anyway i think that triggering a reconcile only if the update event was due to a successful reconciliation would be better here, @rfranzke @timuthy wdyt?

Reply:
Any updates here?